service/client_state.hh

```diff
@@ -126,6 +128,18 @@ private:
 
     workload_type _workload_type = workload_type::unspecified;
 
+    class qos_configuration_change_client_state_subscriber : public qos::qos_configuration_change_subscriber {
```

Comment:
You could make client_state inherit privately from configuration_change_subscriber.

But if you're using indirection, let's squeeze some compile time decoupling and forward-declare the class here, and define it in .cc, so you don't have to add a #include to .hh.

Reply:
Removed the subscriber class and made client state inherit privately from configuration_change_subscriber